From 5f853e5185462f614eefa9cb87f0771f69cc9783 Mon Sep 17 00:00:00 2001
From: IlyaSergeyev <ilya.sergeev@intel.com>
Date: Fri, 9 Sep 2022 13:29:40 +0800
Subject: [PATCH 4/4] FROMGIT: iHD: [VP9][Encode] Unaligned height corruption
 w/a fix

* [VP9][Encode] Unaligned height corruption w/a fix

limit change to 420 surfaces only

(cherry picked from commit 7d187310abd92192babf8c4e5c7b9a770222d5c9 https://github.com/intel/media-driver.git)

BUG=b:231639580
TEST= emerge- libva-intel-media-driver

Signed-off-by: Gareth Yu <gareth.yu@intel.com>
---
 .../agnostic/common/codec/hal/codechal_vdenc_vp9_base.cpp       | 2 +-
 .../agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp         | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/media_driver/agnostic/common/codec/hal/codechal_vdenc_vp9_base.cpp b/media_driver/agnostic/common/codec/hal/codechal_vdenc_vp9_base.cpp
index 980f3738cca9..d50db5fd732c 100644
--- a/media_driver/agnostic/common/codec/hal/codechal_vdenc_vp9_base.cpp
+++ b/media_driver/agnostic/common/codec/hal/codechal_vdenc_vp9_base.cpp
@@ -4336,7 +4336,7 @@ MOS_STATUS CodechalVdencVp9State::SetHcpSrcSurfaceParams(MHW_VDBOX_SURFACE_PARAM
 
     CODECHAL_ENCODE_FUNCTION_ENTER;
 
-    if (MEDIA_IS_WA(m_waTable, Wa_Vp9UnalignedHeight))
+    if (MEDIA_IS_WA(m_waTable, Wa_Vp9UnalignedHeight) && (m_rawSurfaceToPak->Format == Format_NV12 || m_rawSurfaceToPak->Format == Format_P010))
     {
         m_rawSurfaceToPak->dwHeight = m_oriFrameHeight;
     }
diff --git a/media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp b/media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp
index 3e9bf2293e56..f9de370e90fd 100644
--- a/media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp
+++ b/media_driver/agnostic/gen12/codec/hal/codechal_vdenc_vp9_g12.cpp
@@ -3461,7 +3461,6 @@ void CodechalVdencVp9StateG12::fill_pad_with_value(PMOS_SURFACE psSurface)
         return;
     }
 
-    // 4:2:0 only
     if (psSurface->Format == Format_NV12 || psSurface->Format == Format_P010)
     {
         MOS_LOCK_PARAMS lockFlags;
-- 
2.31.0

